tests/ops/test_stats.py

```diff
@@ -116,3 +194,20 @@ def test_diagnostics_ok_with_sample_shape(diagnostics, sample_shape):
         assert_equal(diagnostics(a, chain_dim=1, sample_dim=0), y)
         assert_equal(diagnostics(b, chain_dim=-1, sample_dim=0), y)
         assert_equal(diagnostics(c, sample_dim=-1), y)
+
+
+def test_waic():
+    x = - torch.arange(1., 101).log().reshape(25, 4)
```

Comment:
Are there any restrictions to the shape of `obs_nodes`? Can we test that other arbitrary shapes do not throw an error?

Reply:
There is no restriction for it. I'll make a test soon.